src/validation.cpp

```diff
@@ -3101,10 +3101,12 @@ bool CheckBlock(const CBlock& block, CValidationState& state, const Consensus::P
             return state.DoS(100, false, REJECT_INVALID, "bad-cb-multiple", false, "more than one coinbase");
 
     // Check transactions
-    for (const auto& tx : block.vtx)
-        if (!CheckTransaction(*tx, state, true))
-            return state.Invalid(false, state.GetRejectCode(), state.GetRejectReason(),
-                                 strprintf("Transaction check failed (tx hash %s) %s", tx->GetHash().ToString(), state.GetDebugMessage()));
+    for (const auto& tx : block.vtx) {
+        if (!CheckTransaction(*tx, state, true)) {
+            LogPrintf("Transaction check failed (tx hash %s) %s\n", tx->GetHash().ToString(), state.GetDebugMessage());
```

Comment:
Two things here:
- I think the point here was to add the specific failed tx hash as context information
- Not sure it's a good idea to add `LogPrintf` in validation. We've spent a lot of time some years ago to remove all direct logging in validation to reduce the spamminess for users (don't want `ERROR`s in the log for things that are not the users's fault), replacing it with messages in the validation state, so that one summary message can be given at most

Reply:
Thanks, I see your point (and good catch on the unconditional log message, I agree that we should not do that). I'll revert this change.